backend/persistence/webauthn_credential_persister.go

```diff
@@ -47,6 +47,17 @@ func (p *webauthnCredentialPersister) Create(credential models.WebauthnCredentia
 		return fmt.Errorf("credential object validation failed: %w", vErr)
 	}
 
+	// Eager creation seems to be broken, so we need to store the transports separately.
+	// See: https://github.com/gobuffalo/pop/issues/608
+	vErr, err = p.db.ValidateAndCreate(&credential.Transports)
+	if err != nil {
+		return fmt.Errorf("failed to store credential transport: %w", err)
+	}
+
+	if vErr != nil && vErr.HasAny() {
+		return fmt.Errorf("credential transport validation failed: %w", vErr)
+	}
+
```

Comment:
I don't know if i like this, because what happens when inserting the transports fails, then the credential itself will be stored in the DB, but not the transports and an error will be returned and most likely the endpoint which called this function will retune an error too. So the outcome of this is, that we have an (maybe) unused credential in the DB without any transports, although an error was returned.

I know, that this function is currently called only once and it is wrapped in a database transaction, but if we use this function elsewhere in the future we might have forgotten this behaviour.

I think we should either separate storing the credential and its transports in two different functions or at least add a comment to this function that if should only be called in a database transaction.

Reply:
I don't like it myself. Had to do it this way because eager creation doesn't work. The point about needing a transaction is of course true, but that doesn't change if I create two different methods. Also currently a transaction is used. I thought about splitting the method in two parts, but where to put the new method? Would I create a new persister, because one persister takes care of one entity type?

Reply:
After thinking more about that, we should leave it like it is (for now) but add a comment to the `Create()` function, that it should only be called inside a database transaction.